xc/xenops_server_xen.ml

```diff
@@ -1241,8 +1247,14 @@ module VM = struct
             begin
               debug "Renaming domain %d from %s to %s" di.Xenctrl.domid vm vm';
               Xenctrl.domain_sethandle xc di.Xenctrl.domid vm';
-              let final_uuid_path = "/vm/" ^ vm ^ "/final-uuid" in
-              safe_rm xs final_uuid_path;
+              ( match during with
+                | Pre_migration ->
+                  let origin_uuid_path = "/vm/" ^ vm' ^ "/origin-uuid" in
```

Comment:
use sprintf?

Reply:
I'm following the same style it was being used for now. I'll clean up the whole function after the fix works, there are more changes I want to do, like using `Option.iter` instead of a match